config/acme/machines/config_batch.xml

```diff
@@ -454,8 +461,10 @@
     <job name="case.st_archive">
       <template>template.st_archive</template>
       <task_count>1</task_count>
-      <!-- If DOUT_S is true and case.run (or case.test) exits successfully then run st_archive-->
-      <dependency>case.run or case.test</dependency>
+      <!-- If DOUT_S is true and case.run amd/or case.test exits (regardless of success) then run st_archive
+	   This dependence is 'and' if both are run before case.st, and is 'or' if one is not run
+      -->
+      <dependency>case.run case.test</dependency>
```

Comment:
The original logic completely ignored the "or"; as it was looking for job names. While this change wasn't necessary for this PR; implying functionality that we don't (and can't, AFAICT most batch systems don't support OR in dependencies) have seems bad to me.